src/main/java/uk/gov/hmcts/reform/sendletter/config/RedisConfiguration.java

```diff
@@ -1,26 +1,40 @@
 package uk.gov.hmcts.reform.sendletter.config;
 
 import org.redisson.Redisson;
-import org.redisson.api.RedissonClient;
 import org.redisson.config.Config;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import uk.gov.hmcts.reform.sendletter.cache.SentLettersCache;
+import uk.gov.hmcts.reform.sendletter.services.LetterChecksumGenerator;
 
 @Configuration
 public class RedisConfiguration {
 
     @Bean
-    public RedissonClient redisClient(
+    @ConditionalOnProperty(name = "redis.enabled", havingValue = "true")
+    public SentLettersCache sentLettersCache(
         @Value("${redis.host}") String host,
         @Value("${redis.port}") int port,
-        @Value("${redis.password}") String password
+        @Value("${redis.password}") String password,
+        @Value("${ttlInSeconds}") long ttlInSeconds
     ) {
         Config config = new Config();
         config.useSingleServer()
             .setAddress(host + ":" + port)
             .setPassword(password);
 
-        return Redisson.create(config);
+        return new SentLettersCache(
+            Redisson.create(config),
+            new LetterChecksumGenerator(),
+            ttlInSeconds
+        );
+    }
+
+    @Bean
+    @ConditionalOnProperty(name = "redis.enabled", havingValue = "false")
+    public SentLettersCache sentLettersDisabledCache() {
+        return new SentLettersCache();
```

Comment:
I am not sure if this is a good sign when flags begin to creep in and modify behaviour. What about an alternative approach with interface and two implementations? Both `SentLettersRedisCache` and `SentLettersNullCache` would implement `SentLettersCache`. Actual decision about cache behaviour is made upon instantiation of the object and not on runtime. Not to mention cleaner and simpler code in both implementations.